client/python/gradio_client/client.py

```diff
@@ -0,0 +1,320 @@
+"""The main Client class for the Python client."""
+from __future__ import annotations
+
+import concurrent.futures
+import json
+import re
+import uuid
+from concurrent.futures import Future
+from typing import Callable, Dict, List
+
+import requests
+import websockets
+from gradio_client import serializing, utils
+from packaging import version
+
+
+class Client:
+    def __init__(
+        self,
+        space: str | None = None,
+        src: str | None = None,
+        access_token: str | None = None,
+        max_workers: int = 40,
+    ):
+        self.access_token = access_token
+        self.headers = (
+            {"Authorization": "Bearer {access_token}"} if access_token else {}
+        )
```

Comment:
Would be a could use case for `huggingface_hub` here:

```py
from huggingface_hub.utils import build_hf_headers

headers = build_hf_headers(token=access_token)
# or even
headers = build_hf_headers(token=access_token, library_name="gradio_client", library_version"0.xx")
```
Especially helpful if the user has used `huggingface-cli login` or `huggingface_hub.login` (+it adds a user-agent if you are interested in telemetry)